packages/ui-library/src/components/sidebar-navigation/icon.js

```diff
@@ -0,0 +1,28 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React from "react";
+
+/**
+ * @param {JSX.ElementClass} [as="span"] The component.
+ * @param {?JSX.node} [children=null] The content.
+ * @param {string} [className] The classname.
+ * @param {Object} [props] Extra props.
+ * @returns {JSX.Element} The element.
+ */
+export const Icon = ( { as: Component = "span", children = null, className, ...props } ) => {
```

Comment:
I didn't want to make `children` a required prop. And then I thought it was nice to make the default explicit. And for some reason we don't like explicit `undefined`. I suppose it can be an inexplicit `undefined` just fine.

Looking at the UI library code, we seem to have all these variants in places. Even default to empty `string` in the FileImport.